libcxx/test/std/iterators/iterator.requirements/iterator.cust/robust_against_ordinary_unqualified_lookup.compile.pass.cpp

```diff
@@ -0,0 +1,28 @@
+//===----------------------------------------------------------------------===//
+//
```

Comment:
i would suggest to add another test to test the use case in the paper


```
class Test {
    friend size_t size(const Test&) {
        return 0;
    }
};

size_t f(Test t) {
   return std::ranges::size(t);
}
```
without implementing the paper the above code does not work because the poison pill was too toxic. and with the implementation the above code should work.  I suggest to add
robust_against_poison_pill  which tests all the CPOs